owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt

```diff
@@ -0,0 +1,166 @@
+
+        every { ocAppRegistryService.getAppRegistry() } returns getAppRegistryResult
+
+        val appRegistry = ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+
+        assertEquals(appResgitryMock, appRegistry)
+
+        verify(exactly = 1) { ocAppRegistryService.getAppRegistry() }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getAppRegistryForAccount returns an Exception when the operation is not successful`() {
+        val getAppRegistryResult: RemoteOperationResult<AppRegistryResponse> =
+            createRemoteOperationResultMock(data = appResgitryResponse, isSuccess = false)
+
+        every { ocAppRegistryService.getAppRegistry() } returns getAppRegistryResult
+
+        ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+
+    }
+
+    @Test(expected = Exception::class)
+    fun `getAppRegistryForAccount returns an Exception when getAppRegistry() has an error controlled by an Exception`() {
+        every { ocAppRegistryService.getAppRegistry() } throws Exception()
+
+        ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+    }
+
+    @Test
+    fun `getUrlToOpenInWeb returns an url to open in website`() {
+        val expectedUrl = "https://example.com/file123/TestApp"
+        val getUrlToOpenInWebResult: RemoteOperationResult<String> = createRemoteOperationResultMock(data = expectedUrl, isSuccess = true)
+
+        every {
+            ocAppRegistryService.getUrlToOpenInWeb(
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
+            )
+        } returns getUrlToOpenInWebResult
+
+        val result = ocAppRegistryService.getUrlToOpenInWeb(
+            openWebEndpoint = openWebEndpoint,
+            fileId = OC_FILE.remoteId.toString(),
+            appName = appName
+        )
+
+        assertEquals(expectedUrl, result.data)
+
+        verify {
+            ocAppRegistryService.getUrlToOpenInWeb(
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
+            )
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getUrlToOpenInWeb returns an exception when something there is an error in the method`() {
+
+        every {
+            ocAppRegistryService.getUrlToOpenInWeb(
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
+            )
+        } throws Exception()
+
+        ocAppRegistryService.getUrlToOpenInWeb(
+            openWebEndpoint = openWebEndpoint,
+            fileId = OC_FILE.remoteId.toString(),
+            appName = appName
+        )
+    }
+
+    @Test
+    fun `createFileWithAppProvider returns the url to open in web`() {
+
+        val expectedFileUrl = "https://example.com/files/testFile.txt"
```

Comment:
There is no constant defined to use here? If there isn't I will create e new one.